chainer/functions/cross_covariance.py

```diff
@@ -0,0 +1,109 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class CrossCovariance(function.Function):
+
+    """Cross-covariance loss."""
+
+    def __init__(self, use_cudnn=True):
+        self.use_cudnn = use_cudnn
+        self.y_centered = None
+        self.z_centered = None
+        self.covariance = None
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        z_type, y_type = in_types
+
+        type_check.expect(
+            z_type.dtype == numpy.float32,
+            z_type.ndim == 2,
+            y_type.dtype == numpy.float32,
+            y_type.ndim == 2,
+
+            z_type.shape[0] == y_type.shape[0]
+        )
+
+    def check_type_backward(self, in_types, out_types):
+        type_check.expect(
+            in_types.size() == 2,
+            out_types.size() == 1,
+        )
+        y_in_type, z_in_type = in_types
+        out_type, = out_types
+
+        type_check.expect(out_type.dtype == y_in_type.dtype)
+        type_check.expect(out_type.dtype == z_in_type.dtype)
+
+    def forward_cpu(self, inputs):
+        y, z = inputs
+        y_mean = y.mean(axis=0, keepdims=True)
+        z_mean = z.mean(axis=0, keepdims=True)
+        self.y_centered = (y - y_mean)[:, :, None]
+        self.z_centered = (z - z_mean)[:, None, :]
+        self.covariance = (self.y_centered * self.z_centered).mean(axis=0)
+        cost = 0.5 * (self.covariance**2).sum(keepdims=True)
+        return cost.reshape(()),
```

Comment:
If I remove ``keepdims`` ``cost`` will be of type float (not a ndarray anymore). To quote from http://docs.scipy.org/doc/numpy/reference/generated/numpy.sum.html#numpy.sum 
*if a is a 0-d array, or if axis is None, a scalar is returned*